### ISSUE TITLE
sslverify code uses strings instead of bytes, which is problem on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8672]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8672|
|Type|enhancement|
|Created|2016-07-25 10:17:31Z|

If I do the following:

1.  Create a python3 virtual environment:

```
python -m venv test_venv1
```

2.  Source it:

```
. test_venv1/bin/activate
```

3.  Install pyOpenSSL:

```
pip install pyOpenSSL
```

4.  Checkout the code:

```
git clone https://github.com/twisted/twisted twisted_test1
```

5.  Run the tests, with additional debugging:

```
python -Wall -bb bin/trial twisted.test.test_ssl
python -Wall -bb bin/trial twisted.test.test_sslverify
```

There are some DeprecationWarnings where str is used,
when bytes should really be used.  For example:

```
twisted/internet/_sslverify.py:1658: DeprecationWarning: str for buf is no longer accepted, use bytes
twisted/internet/_sslverify.py:1662: DeprecationWarning: str for cipher_list is no longer accepted, use bytes
twisted/internet/_sslverify.py:1801: DeprecationWarning: str for cipher_list is no longer accepted, use bytes
```

This code needs to be cleaned up so that these warnings are eliminated.

<details><summary>Searchable metadata</summary>

```
trac-id__8672 8672
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469441851210249 1469441851210249
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_1"></a>|@acabhishek942 commented|
|-|-|

See https://github.com/twisted/twisted/pull/415

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 99137f53a8a6bd1d8bffa5202373236162daf94e
```
#!CommitTicketReference repository="" revision="99137f53a8a6bd1d8bffa5202373236162daf94e"
Merge pull request #425 from twisted/8672-acabhishek942-bytes

Author: acabhishek942
Reviewer: rodrigc
Fixes: #8672

Use bytes instead of string in twisted.internet._sslverify
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
